extensions/podman/src/podman-install.ts

```diff
@@ -294,3 +310,97 @@ class WinInstaller extends BaseInstaller {
     });
   }
 }
+
+abstract class BaseCheck implements extensionApi.InstallCheck {
+  abstract title: string;
+  abstract execute(): Promise<extensionApi.CheckResult>;
+
+  protected createFailureResult(description?: string): extensionApi.CheckResult {
+    return { successful: false, description };
+  }
+
+  protected createSuccessfulResult(): extensionApi.CheckResult {
+    return { successful: true };
+  }
+}
+
+class WinBitCheck extends BaseCheck {
+  title = 'Windows x64';
+  private ARCH = 'x64';
+  async execute(): Promise<extensionApi.CheckResult> {
+    const currentArch = process.arch;
+    if (this.ARCH === currentArch) {
+      return this.createSuccessfulResult();
+    } else {
+      return this.createFailureResult('WSL2 works only on x64 OS.');
+    }
+  }
+}
+
+class WinVersionCheck extends BaseCheck {
+  title = 'Windows Version';
+
+  private MIN_BUILD = 18362;
+  async execute(): Promise<extensionApi.CheckResult> {
+    const winRelease = os.release();
+    if (winRelease.startsWith('10.0.')) {
+      const splitRelease = winRelease.split('.');
+      const winBuild = splitRelease[2];
+      if (Number.parseInt(winBuild) >= this.MIN_BUILD) {
+        return { successful: true };
+      } else {
+        return this.createFailureResult(
+          'To be able to run WSL2 you need Windows 10 Build 18362 or later. Please update your OS.',
```

Comment:
add link to
https://docs.microsoft.com/en-us/windows/wsl/install-manual#step-2---check-requirements-for-running-wsl-2